mitiq/pec/representations/learning.py

```diff
@@ -172,28 +325,60 @@ def biased_noise_loss_function(
     """
     epsilon = params[0]
     eta = params[1]
-    representations = [
-        represent_operation_with_local_biased_noise(
-            operation,
-            epsilon,
-            eta,
-        )
-        for operation in operations_to_mitigate
-    ]
-    mitigated_values = np.array(
-        [
-            execute_with_pec(
-                circuit=training_circuit,
-                observable=observable,
-                executor=noisy_executor,
-                representations=representations,
-                full_output=False,
-                **pec_kwargs,
+
+    if pec_data is not None:
+        ind_eps = np.abs(pec_data[:, 0, 0] - epsilon).argmin()
+        ind_eta = np.abs(pec_data[0, :, 0] - eta).argmin()
+        mitigated_values = pec_data[ind_eps, ind_eta, :]
```

Comment:
I am not sure why, but on the depolarizing noise function you skip the first element of the array with 
 `mitigated_values = pec_data[ind, 1:]`. 

I wonder why here we don't need to skip, e.g. something like:

`        mitigated_values = pec_data[ind_eps, ind_eta, 1:]
`

Reply:
Good question, I had to go back and check this. The indexing is intentionally slightly different for depolarizing noise versus biased noise. 

In the depolarizing noise loss function, the first column contains values of epsilon and the remaining columns are PEC-mitigated expectation values, one column per training circuit. That's why I skip the first element with `mitigated_values = pec_data[ind, 1:]`

In the biased noise loss function, each "layer" along the third dimension corresponds to a training circuit. So, once we've identified the value of noise strength (row) and noise bias (column), we take the values from all the training circuits, i.e. `mitigated_values = pec_data[ind_eps, ind_eta, :]`.